src/Plugins/ComplexCore/src/ComplexCore/Filters/ImportHDF5Dataset.cpp

```diff
@@ -24,9 +24,8 @@ std::vector<size_t> createDimensionVector(const std::string& cDimsStr)
 {
   std::vector<size_t> cDims;
   std::vector<std::string> dimsStrVec = StringUtilities::split(cDimsStr, std::vector<char>{','}, true);
-  for(int i = 0; i < dimsStrVec.size(); i++)
+  for(auto dimsStr : dimsStrVec)
```

Comment:
Instead of making this non-const ref, the output of `trimmed` can just be set to a new variable.